### PR TITLE
Adding ONNX support for ImageGPT

### LINKED ISSUE #555: Community contribution -  `optimum.exporters.onnx` support for new models!
Following what was done by @ChainYo in Transformers, in the [ONNXConfig: Add a configuration for all available models](https://github.com/huggingface/transformers/issues/16308) issue, the idea is to add support for exporting new models in `optimum.exporters.onnx`.

This issue is about the working group specially created for this task. If you are interested in helping out, reply here, take a look at this [organization](https://huggingface.co/OWG), or add  `ChainYo#3610` on discord.

We want to contribute to Hugging Face's ONNX export implementation for all available models on Hugging Face Hub. There are already a lot of architectures implemented for converting PyTorch models to ONNX, but we need more! We need them all!

Feel free to join us in this adventure! Join the org by clicking [here](https://huggingface.co/organizations/OWG/share/TskjfGaGjGnMXXssbPPXrQWEIbosGqZshZ)

Here is a non-exhaustive list of models that all models available:

- [x] Albert
- [x] BART
- [x] BeiT
- [x] BERT
- [ ] BigBird (Critical issue: https://github.com/huggingface/optimum/issues/754#issuecomment-1429830467)
- [ ] BigBirdPegasus (Critical issue: https://github.com/huggingface/optimum/issues/754#issuecomment-1429830467)
- [x] Blenderbot
- [x] BlenderbotSmall
- [ ] BLIP-2
- [x] BLOOM
- [x] CamemBERT
- [ ] CANINE
- [x] CLIP
- [x] CodeGen
- [x] ConvNext
- [x] ConvBert
- [ ] CTRL
- [x] CvT
- [x] Data2VecText
- [x] Data2VecVision
- [x] Deberta
- [x] DebertaV2
- [x] DeiT
- [ ] DecisionTransformer
- [x] DETR
- [x] Distilbert
- [ ] DPR
- [x] DPT
- [x] ELECTRA
- [ ] FNet
- [ ] FSMT
- [x] Flaubert
- [ ] FLAVA
- [ ] Funnel Transformer
- [x] GLPN
- [x] GPT2
- [x] GPTJ
- [x] GPT-Neo
- [x] GPT-NeoX
- [x] Hubert
- [x] I-Bert
- [x] ImageGPT 🛠️ @adit299 
- [ ] LED
- [x] LayoutLM
- [ ] LayoutLMv2 (but 🛠️ in Transformers)
- [x] LayoutLMv3
- [ ] LayoutXLM
- [ ] LED
- [x] LeViT
- [ ] 🛠️ Longformer (Critical issue: https://github.com/huggingface/optimum/issues/776#issuecomment-1429680121)
- [x] LongT5
- [ ] Luke (but 🛠️ in Transformers)
- [ ] Lxmert
- [x] M2M100
- [ ] MaskFormer
- [x] mBart
- [ ] MCTCT
- [x] MPNet
- [x] MT5
- [x] MarianMT
- [ ] MegatronBert
- [x] MobileBert
- [x] MobileViT
- [ ] Nyströmformer
- [x] OpenAIGPT-2
- [x] OPT ((but 🛠️ in Transformers)
- [x] OWLViT
- [x] Pix2Struct
- [x] PLBart
- [x] Pegasus
- [x] Perceiver
- [x] PoolFormer
- [ ] ProphetNet
- [ ] QDQBERT
- [ ] RAG
- [ ] REALM
- [ ] Reformer (but 🛠️ in Transformers)
- [x] RemBert
- [x] ResNet
- [x] RegNet 🛠️ @asrimanth 
- [ ] RetriBert
- [x] RoFormer
- [x] RoBERTa
- [x] SEW
- [x] SEW-D
- [x] SegFormer
- [x] Speech2Text
- [ ] Speech2Text2
- [x] Splinter
- [x] SqueezeBERT
- [x] Swin Transformer
- [x] T5
- [ ] TAPAS 🛠️ @someshfengde 
- [ ] TAPEX
- [ ] Transformer XL
- [x] TrOCR
- [ ] UniSpeech
- [ ] UniSpeech-SAT
- [ ] VAN
- [x] ViT
- [ ] Vilt
- [ ] VisualBERT
- [x] Wav2Vec2
- [x] WavLM
- [x] Whisper
- [ ] XGLM
- [x] XLM
- [ ] XLMProphetNet
- [x] XLM-RoBERTa
- [x] XLM-RoBERTa-XL
- [ ]  XLNet (but 🛠️ in Transformers)
- [x] YOLOS
- [ ] Yoso

🛠️ next to a model suggests that the PR is in progress. If there is nothing next to a model, it means that ONNX does not yet support the model, and thus we need to add support for it.

If you need help implementing an unsupported model, here is a [guide](https://huggingface.co/docs/optimum/main/en/exporters/onnx/usage_guides/contribute) from HuggingFace Optimum documentation.

### PR DESCRIPTION
**Pull request by adit299**
# What does this PR do?

Fixes #555 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Comment by adit299** [comment]
Hello, I believe I have completed adding ONNX support for ImageGPT. I ran the tests mentioned here (but using imagegpt instead of resnet) and they pass:
> Thank you @asrimanth ! You can try:
> 
> ```
> pip install -e .[tests,exporters]
> pytest tests/exporters/onnx/test_*.py -k "regnet" -s
> ```
> 
> For slow tests:
> 
> ```
> pip install -e .[tests,exporters]
> RUN_SLOW=1 pytest tests/exporters/onnx/test_*.py -k "regnet" -s -m "run_slow"
> ```
Please do let me know if everything is good with this PR!

File: optimum/exporters/onnx/model_configs.py [hunk 1]
```diff
@@ -1035,8 +1035,7 @@ def inputs(self) -> Dict[str, Dict[int, str]]:
 
             if self.use_past_in_inputs:
                 self.add_past_key_values(common_inputs, direction="inputs")
-
         if self._behavior is ConfigBehavior.DECODER:
             common_inputs["encoder_outputs"] = {0: "batch_size", 1: "encoder_sequence_length"}
-
+
```
Comment (regisss) [inline optimum/exporters/onnx/model_configs.py:1040]: The code quality test didn't pass. Can you run the command `make style` in the root of the Optimum repo (you'll need to have the right packages installed with `pip install -e .[dev,exporters]`)?

File: tests/exporters/exporters_utils.py [hunk 2]
```diff
@@ -59,6 +59,7 @@
     "gptj": "hf-internal-testing/tiny-random-GPTJModel",
     "groupvit": "hf-internal-testing/tiny-random-groupvit",
     "ibert": "hf-internal-testing/tiny-random-IBertModel",
+    "imagegpt": "hf-internal-testing/tiny-random-ImageGPTModel",
```
Comment (regisss) [inline tests/exporters/exporters_utils.py:62]: Could you also add an entry below in `PYTORCH_EXPORT_MODELS_LARGE` for [this model](https://huggingface.co/openai/imagegpt-small) for instance?

**Review by regisss — COMMENTED** [review]
Hi @adit299! It looks very good, just a few more things to do and we will be able to merge it :)

File: tests/exporters/exporters_utils.py (hunk 2 above)
Comment (adit299) [reply tests/exporters/exporters_utils.py:62]: fixed

**Review by fxmarty — APPROVED** [review]
Thank you for the addition!

**Pull request merged**